### ISSUE TITLE
Accessibility: Selected buttons cannot be triggered on Game Controller

### ISSUE THREAD

**Issue by creativecreatorormaybenot** [flutter/flutter#47697]
## Steps to Reproduce

When using a Flutter app with a POWKIDDY X18 device, you can select buttons fine (which is the same as pressing <kbd>Tab</kbd> on a keyboard in the Android Emulator), but you cannot trigger them (which would be pressing <kbd>Enter</kbd>).

**Target Platform: Android**
**Devices: POWKIDDY X18**

## Expected behavior

In native Android apps, the **`A`** game button can be pressed to trigger selected buttons, but in Flutter apps it does not work.  

## Logs

```
flutter doctor -v
[√] Flutter (Channel master, v1.13.3-pre.23, locale en-US)
    • Flutter version 1.13.3-pre.23 at flutterRoot
    • Framework revision c06bf6503a (10 days ago), 2019-12-13 17:42:35 -0500
    • Engine revision e0e0ac0a68
    • Dart version 2.8.0 (build 2.8.0-dev.0.0 45db297095)
```

**Comment by creativecreatorormaybenot** [comment]
@gspencergoog Is it possible that this is related to https://github.com/flutter/flutter/pull/33868?

**Comment by iapicca** [comment]
Hi @creativecreatorormaybenot 
Even if flutter had an "out-of-the-box" game controller support,
this might not the use case;

I'm not sure that said product has an [sdk](https://github.com/retrogamehandheld/powkiddyx18/wiki),
but I'd like to share the experience I had dealing with these [devices](https://www.zebra.com/gb/en/cpn/os-migration/explore-our-solutions/android-home.html).

I just linked the "button event" Android side 
to a Platform Channel and everything worked seamlessly.

<details>
	<summary>android</summary>

```java
package com.example.##################;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.IntentFilter;
import android.os.Bundle;
import io.flutter.app.FlutterActivity;
import io.flutter.plugin.common.EventChannel;
import io.flutter.plugins.GeneratedPluginRegistrant;



public class MainActivity extends FlutterActivity {

  static IntentFilter filter = new IntentFilter();
  static String mAction = "###############";
  static final String mKey = "###############";
  static final String STREAM = "## flutter side ##/Stream";
  EventChannel.EventSink mEventSink;

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    GeneratedPluginRegistrant.registerWith(this);
    filter.addCategory(Intent.CATEGORY_DEFAULT);
    filter.addAction(mAction);
    registerReceiver(myBroadcastReceiver, filter);
    new EventChannel(getFlutterView(), STREAM).setStreamHandler(
            new EventChannel.StreamHandler() {
              @Override
              public void onListen(Object args, final EventChannel.EventSink eventSink) {
                mEventSink=eventSink;
              }
              @Override
              public void onCancel(Object args) {
                mEventSink=null;
              }
            }
    );
  }

  @Override
  protected void onDestroy() {
    super.onDestroy();
    unregisterReceiver(myBroadcastReceiver);
  }

  private BroadcastReceiver myBroadcastReceiver = new BroadcastReceiver() {
    @Override
    public void onReceive(Context context, Intent intent) {
      String action = intent.getAction();
      if (action.equals(mAction)) {
        try {
          String data = intent.getStringExtra(mKey);
          mEventSink.success(data);
        } catch (Exception e) {
        }
      }
    }
  };
}
```

</details>

<details>
	<summary>flutter</summary>

```dart
  static const String _eventStream = '## flutter side ##/Stream';
  static const EventChannel eventChannel = const EventChannel(_eventStream);
  static StreamSubscription _streamSubscription;
  Future<bool> enableScanner({final Information info}) async {
    await disableScanner();
    _streamSubscription = eventChannel.receiveBroadcastStream()
      .listen((data) {
        /// #### YOUR LOGIC ####
      });
    return true;
  }

  Future<bool> disableScanner() async {
    await _streamSubscription?.cancel();
    _streamSubscription = null;
    return true;
  }
```

</details>

but of course you need to know the `intent` first.
I hope you find this helpful.
thank you

**Comment by creativecreatorormaybenot** [comment]
@iapicca I think that it is actually possible to achieve what you did using Android native code using only Flutter code because of https://github.com/flutter/flutter/pull/33868.

See the following line:

https://github.com/gspencergoog/flutter/blob/679b6e2999c1c53ef57e5db4903fb0e2f35d96c9/packages/flutter/lib/src/services/keyboard_maps.dart#L182

This can be handled in [`FocusNode.onKey`](https://api.flutter.dev/flutter/widgets/FocusNode/onKey.html).

### Actual problem

If my assumption is correct, the actual problem here would be that the accessibility functionality in Flutter does not handle this "*game button A*" press, even though it should do so. As I explained previously, pressing the <kbd>Enter</kbd> button on a physical keyboard **will** trigger this accessibility action and **game button A** should do so as well, but I could not find the relevant section in the Flutter SDK.

**Comment by gspencergoog** [comment]
I think this may simply be because the `ActivateAction` isn't bound to the game controller A button.

If you add a `Shortcuts` widget around your app with the following shortcut map, does it do what you expect?

```dart
Shortcuts(
  shortcuts: <LogicalKeySet, Intent>{LogicalKeySet(LogicalKeyboardKey.gameButtonA): const Intent(ActivateAction.key)}
  child: ...)
```

**Comment by creativecreatorormaybenot** [comment]
@gspencergoog Thanks! That is what I was looking for and also what I meant.  
I will post an update once I know this works for certain.

Do you think having this link by default would be an option?

**Comment by gspencergoog** [comment]
Yes, that's definitely an option, but let's make sure it fixes the problem first.

**Comment by creativecreatorormaybenot** [comment]
@gspencergoog I can confirm that it works using `Shortcuts`. Navigation already worked before using the D-pad as I explained previously.

I assume that there is some lower level definition of shortcuts within the framework, but I am not sure where they are defined - otherwise I could open a PR.

Anyways, I am going to close this issue for now - I could open a new one for a feature request to make this a default if needed.
